From: Muhammad Usama Anjum <usama.anjum@collabora.com>
Subject: selftests/mm: add -Wunused family of flags
Date: Thu, 31 Jul 2025 21:01:28 +0500

Add -Wunused family of flags and fix all the warnings coming because of
argc and argv.  Remove them if they aren't being used entirely.  Use
__unused compiler attribute with argc where argv is being used.

Link: https://lkml.kernel.org/r/20250731160132.1795351-5-usama.anjum@collabora.com
Signed-off-by: Muhammad Usama Anjum <usama.anjum@collabora.com>
Cc: Albert Ou <aou@eecs.berkeley.edu>
Cc: Alexandre Ghiti <alex@ghiti.fr>
Cc: Baolin Wang <baolin.wang@linux.alibaba.com>
Cc: Barry Song <baohua@kernel.org>
Cc: David Hildenbrand <david@redhat.com>
Cc: Dev Jain <dev.jain@arm.com>
Cc: Jason Gunthorpe <jgg@ziepe.ca>
Cc: John Hubbard <jhubbard@nvidia.com>
Cc: Leon Romanovsky <leon@kernel.org>
Cc: Liam Howlett <liam.howlett@oracle.com>
Cc: Lorenzo Stoakes <lorenzo.stoakes@oracle.com>
Cc: Mariano Pache <npache@redhat.com>
Cc: Michal Hocko <mhocko@suse.com>
Cc: Mike Rapoport <rppt@kernel.org>
Cc: Palmer Dabbelt <palmer@dabbelt.com>
Cc: Paul Walmsley <paul.walmsley@sifive.com>
Cc: Peter Xu <peterx@redhat.com>
Cc: Ryan Roberts <ryan.roberts@arm.com>
Cc: Shuah Khan <shuah@kernel.org>
Cc: Sidhartha Kumar <sidhartha.kumar@oracle.com>
Cc: Suren Baghdasaryan <surenb@google.com>
Cc: Vlastimil Babka <vbabka@suse.cz>
Cc: Zi Yan <ziy@nvidia.com>
Cc: Kevin Brodsky <kevin.brodsky@arm.com>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 tools/testing/selftests/mm/Makefile                |    2 +-
 tools/testing/selftests/mm/compaction_test.c       |    2 +-
 tools/testing/selftests/mm/cow.c                   |    2 +-
 tools/testing/selftests/mm/droppable.c             |    2 +-
 tools/testing/selftests/mm/gup_longterm.c          |    2 +-
 tools/testing/selftests/mm/hugepage-vmemmap.c      |    2 +-
 tools/testing/selftests/mm/hugetlb-madvise.c       |    2 +-
 tools/testing/selftests/mm/hugetlb-soft-offline.c  |    2 +-
 tools/testing/selftests/mm/madv_populate.c         |    2 +-
 tools/testing/selftests/mm/map_populate.c          |    2 +-
 tools/testing/selftests/mm/memfd_secret.c          |    2 +-
 tools/testing/selftests/mm/mlock-random-test.c     |    2 +-
 tools/testing/selftests/mm/mlock2-tests.c          |    2 +-
 tools/testing/selftests/mm/on-fault-limit.c        |    2 +-
 tools/testing/selftests/mm/pkey_sighandler_tests.c |    2 +-
 tools/testing/selftests/mm/soft-dirty.c            |    2 +-
 tools/testing/selftests/mm/uffd-wp-mremap.c        |    2 +-
 tools/testing/selftests/mm/virtual_address_range.c |    2 +-
 18 files changed, 18 insertions(+), 18 deletions(-)

--- a/tools/testing/selftests/mm/compaction_test.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/compaction_test.c
@@ -203,7 +203,7 @@ int set_zero_hugepages(unsigned long *in
 	return ret;
 }
 
-int main(int argc, char **argv)
+int main(void)
 {
 	struct rlimit lim;
 	struct map_list *list = NULL, *entry;
--- a/tools/testing/selftests/mm/cow.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/cow.c
@@ -1857,7 +1857,7 @@ static int tests_per_non_anon_test_case(
 	return tests;
 }
 
-int main(int argc, char **argv)
+int main(void)
 {
 	struct thp_settings default_settings;
 
--- a/tools/testing/selftests/mm/droppable.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/droppable.c
@@ -15,7 +15,7 @@
 
 #include "../kselftest.h"
 
-int main(int argc, char *argv[])
+int main(void)
 {
 	size_t alloc_size = 134217728;
 	size_t page_size = getpagesize();
--- a/tools/testing/selftests/mm/gup_longterm.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/gup_longterm.c
@@ -504,7 +504,7 @@ static int tests_per_test_case(void)
 	return 3 + nr_hugetlbsizes;
 }
 
-int main(int argc, char **argv)
+int main(void)
 {
 	int i;
 
--- a/tools/testing/selftests/mm/hugepage-vmemmap.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/hugepage-vmemmap.c
@@ -87,7 +87,7 @@ static int check_page_flags(unsigned lon
 	return 0;
 }
 
-int main(int argc, char **argv)
+int main(void)
 {
 	void *addr;
 	unsigned long pfn;
--- a/tools/testing/selftests/mm/hugetlb-madvise.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/hugetlb-madvise.c
@@ -57,7 +57,7 @@ void read_fault_pages(void *addr, unsign
 	}
 }
 
-int main(int argc, char **argv)
+int main(int __unused argc, char **argv)
 {
 	unsigned long free_hugepages;
 	void *addr, *addr2;
--- a/tools/testing/selftests/mm/hugetlb-soft-offline.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/hugetlb-soft-offline.c
@@ -216,7 +216,7 @@ static void test_soft_offline_common(int
 			 enable_soft_offline);
 }
 
-int main(int argc, char **argv)
+int main(void)
 {
 	ksft_print_header();
 	ksft_set_plan(2);
--- a/tools/testing/selftests/mm/madv_populate.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/madv_populate.c
@@ -281,7 +281,7 @@ static int system_has_softdirty(void)
 #endif
 }
 
-int main(int argc, char **argv)
+int main(void)
 {
 	int nr_tests = 16;
 	int err;
--- a/tools/testing/selftests/mm/Makefile~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/Makefile
@@ -20,7 +20,6 @@ endif
 # thus tricking Make (and you!) into believing that All Is Well, in subsequent
 # make invocations:
 .DELETE_ON_ERROR:
-
 # Avoid accidental wrong builds, due to built-in rules working just a little
 # bit too well--but not quite as well as required for our situation here.
 #
@@ -35,6 +34,7 @@ MAKEFLAGS += --no-builtin-rules
 
 CFLAGS = -Wall -O2 -I $(top_srcdir) $(EXTRA_CFLAGS) $(KHDR_INCLUDES) $(TOOLS_INCLUDES)
 CFLAGS += -Wunreachable-code
+CFLAGS += -Wunused  -Wunused-parameter -Wunused-function -Wunused-label -Wunused-variable -Wunused-value
 LDLIBS = -lrt -lpthread -lm
 
 # Some distributions (such as Ubuntu) configure GCC so that _FORTIFY_SOURCE is
--- a/tools/testing/selftests/mm/map_populate.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/map_populate.c
@@ -76,7 +76,7 @@ static int child_f(int sock, unsigned lo
 	return ksft_cnt.ksft_pass;
 }
 
-int main(int argc, char **argv)
+int main(void)
 {
 	int sock[2], child, ret;
 	FILE *ftmp;
--- a/tools/testing/selftests/mm/memfd_secret.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/memfd_secret.c
@@ -299,7 +299,7 @@ static void prepare(void)
 
 #define NUM_TESTS 6
 
-int main(int argc, char *argv[])
+int main(void)
 {
 	int fd;
 
--- a/tools/testing/selftests/mm/mlock2-tests.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/mlock2-tests.c
@@ -425,7 +425,7 @@ static void test_mlockall(void)
 	munlockall();
 }
 
-int main(int argc, char **argv)
+int main(void)
 {
 	int ret, size = 3 * getpagesize();
 	void *map;
--- a/tools/testing/selftests/mm/mlock-random-test.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/mlock-random-test.c
@@ -236,7 +236,7 @@ static void test_mlock_outof_limit(char
 	ksft_test_result_pass("%s\n", __func__);
 }
 
-int main(int argc, char **argv)
+int main(void)
 {
 	char *p = NULL;
 
--- a/tools/testing/selftests/mm/on-fault-limit.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/on-fault-limit.c
@@ -28,7 +28,7 @@ static void test_limit(void)
 	munlockall();
 }
 
-int main(int argc, char **argv)
+int main(void)
 {
 	ksft_print_header();
 	ksft_set_plan(1);
--- a/tools/testing/selftests/mm/pkey_sighandler_tests.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/pkey_sighandler_tests.c
@@ -528,7 +528,7 @@ static void (*pkey_tests[])(void) = {
 	test_pkru_sigreturn
 };
 
-int main(int argc, char *argv[])
+int main(void)
 {
 	int i;
 
--- a/tools/testing/selftests/mm/soft-dirty.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/soft-dirty.c
@@ -194,7 +194,7 @@ static void test_mprotect_file(int pagem
 	test_mprotect(pagemap_fd, pagesize, false);
 }
 
-int main(int argc, char **argv)
+int main(void)
 {
 	int pagemap_fd;
 	int pagesize;
--- a/tools/testing/selftests/mm/uffd-wp-mremap.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/uffd-wp-mremap.c
@@ -334,7 +334,7 @@ static const struct testcase testcases[]
 	},
 };
 
-int main(int argc, char **argv)
+int main(void)
 {
 	struct thp_settings settings;
 	int i, j, plan = 0;
--- a/tools/testing/selftests/mm/virtual_address_range.c~selftests-mm-add-wunused-family-of-flags
+++ a/tools/testing/selftests/mm/virtual_address_range.c
@@ -189,7 +189,7 @@ static int validate_complete_va_space(vo
 	return 0;
 }
 
-int main(int argc, char *argv[])
+int main(void)
 {
 	char *ptr[NR_CHUNKS_LOW];
 	char **hptr;
_
